apps/process-new-granules/src/process_new_granules.py

```diff
@@ -46,6 +46,8 @@ def get_jobs_for_granule(subscription, granule):
             payload.append(job)
     else:
         raise ValueError(f'Subscription job type {job_type} not supported')
+    for job in payload:
+        job['subscription_id'] = subscription['subscription_id']
```

Comment:
I slightly prefer `job_specification['subscription_id'] = subscription['subscription_id']` around line 32, so it's added to the template once up front instead of to each individual job after the fact